projects/eval-ade7754/src/platform/maxim/platform.c

```diff
@@ -0,0 +1,122 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. “AS IS” AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL ANALOG DEVICES, INC. BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include "platform.h"
+
+struct max_uart_init_param max_uart_extra_ip = {
+	.flow = UART_FLOW_DIS
+};
+
+struct max_gpio_init_param gpio_extra = {
+	.vssel = MXC_GPIO_VSSEL_VDDIOH,
+};
+
+struct max_spi_init_param ade7754_spi_extra_ip  = {
+	.num_slaves = SPI_SLAVE_NUM,
+	.polarity = SPI_SS_POL_LOW,
+	.vssel = MXC_GPIO_VSSEL_VDDIOH,
+};
+
+struct max_gpio_init_param ade7754_gpio_extra_ip = {
+	.vssel = MXC_GPIO_VSSEL_VDDIOH,
+};
+
+struct no_os_uart_init_param uart_ip = {
+	.device_id = UART_DEV_ID,
+	.baud_rate = UART_BAUD,
+	.size = NO_OS_UART_CS_8,
+	.parity = NO_OS_UART_PAR_NO,
+	.stop = NO_OS_UART_STOP_1_BIT,
+	.platform_ops = &max_uart_ops,
+	.extra = &max_uart_extra_ip,
+};
+
+/* user led pin */
+struct no_os_gpio_init_param gpio_led1_ip = {
+	.port = GPIO_LED_PORT,
+	.number = GPIO_LED_PIN,
+	.pull = NO_OS_PULL_UP,
+	.platform_ops = &max_gpio_ops,
+	.extra = &gpio_extra,
+};
+
+/* reset pin */
+struct no_os_gpio_init_param gpio_reset_ip = {
+	.port = GPIO_RESET_PORT,
+	.number = GPIO_RESET_PIN,
+	.pull = NO_OS_PULL_NONE,
+	.platform_ops = &max_gpio_ops,
+	.extra = &gpio_extra,
+};
+
+/* irqn pin */
+struct no_os_gpio_init_param ade7754_gpio_irq_ip = {
+	.port = GPIO_IRQ_PORT,
+	.number = GPIO_IRQ_PIN,
+	.pull = NO_OS_PULL_NONE,
+	.platform_ops = &max_gpio_ops,
+	.extra = &gpio_extra,
+};
+
+struct no_os_spi_init_param ade7754_spi_ip = {
+	.device_id = SPI_DEVICE_ID,
+	.max_speed_hz = SPI_BAUDRATE,
+	.bit_order = NO_OS_SPI_BIT_ORDER_MSB_FIRST,
+	.mode = NO_OS_SPI_MODE_1,
+	.platform_ops = &max_spi_ops,
+	.chip_select = SPI_CS,
+	.extra = &ade7754_spi_extra_ip,
+};
+
+struct no_os_irq_init_param ade7754_gpio_int_ip = {
+	.platform_ops = GPIO_IRQ_OPS,
+	.irq_ctrl_id = GPIO_CTRL_IRQ_ID,
+	.extra = GPIO_IRQ_EXTRA,
+};
+
+/**
+ * @brief Toggle Led
+ * @param gpio_led_desc - led descriptor
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int interface_toggle_led(struct no_os_gpio_desc *gpio_led_desc)
+{
+	uint8_t val;
+	int ret;
+
+	ret = no_os_gpio_get_value(gpio_led_desc, &val);
+	if (ret)
+		return ret;
+
+	return no_os_gpio_set_value(gpio_led_desc, !val);
+}
```

Comment:
`common_data.c` or `main.c` are probably better places for this function, since its functionality is not related to the Maxim platform